lib/iris/cube.py

```diff
@@ -3130,18 +3130,19 @@ def _intersect_modulus(
                     pre_wrap_delta != post_wrap_delta
                 )
 
-                # Recalculate the extended minimum.
                 indices = inside_indices[split_cell_indices]
                 cells = bounds[indices]
-                cells_delta = np.diff(coord.bounds[indices])
-
-                # Watch out for ascending/descending bounds
-                if cells_delta[0, 0] > 0:
-                    cells[:, 0] = cells[:, 1] - cells_delta[:, 0]
-                    minimum = np.min(cells[:, 0])
-                else:
-                    cells[:, 1] = cells[:, 0] + cells_delta[:, 0]
-                    minimum = np.min(cells[:, 1])
+                if maximum % modulus not in cells:
+                    # Recalculate the extended minimum.
+                    cells_delta = np.diff(coord.bounds[indices])
+
+                    # Watch out for ascending/descending bounds.
+                    if cells_delta[0, 0] > 0:
```

Comment:
Previously, `minimum` would have been set by either branch of this `if` statement on every call.  Now that it is contained within `if maximum % modulus not in cells` it may not be.  Should there be an `else` to the `if` on L3135?

Reply:
It's only been two months but I had completely forgotten the logic here!  I think this is fine though: we don't explicitly set `minimum` here, so it is just left to whatever `minimum` was originally passed into the method.  Note that because of the previous logic

`maximum - modulus` <= `minimum` (line 3081)
`cells` >= `minimum` (line 3098)

So the only way that `maximum - modulus in cells` will be `True` is if `maximum - modulus` == `minimum` and `minimum` is in `cells`.   I.e. the very specific case described in #3391.  In that case, it is expected that the bounds on the output coordinate will precisely span the requested (minimum, maximum) range.

Definitely needs a comment though!